### ISSUE TITLE
no-use-before-declare error saying that 'undefined' is used before being declared

### ISSUE THREAD

**Issue by rationull** [palantir/tslint#2434]
### Bug Report

- TSLint 5.0.0
- Typescript 2.2.2
- Running TSLint via CLI with --project and --type-check

#### TypeScript code being linted

```ts
function f(x) {
   if (x !== undefined) {
      console.log(x);
   }
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-use-before-declare": true
  }
}

```

#### Actual behavior

Use of 'undefined' should not be flagged because 'undefined' is not a variable.

#### Expected behavior

ERROR is emitted: "variable 'undefined' used before declaration"

**Comment by nchen63** [comment]
fixed in master

**Comment by rationull** [comment]
Thanks @nchen63 for the quick fix!
